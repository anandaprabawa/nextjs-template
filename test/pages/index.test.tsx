import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';
import Home from 'src/pages';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('link element should have attributes', () => {
    const { getByText } = render(<Home />, {});
    const linkEl = getByText('Go to repository');
    expect(linkEl).toHaveAttribute(
      'href',
      'https://github.com/anandaprabawa/nextjs-template',
    );
    expect(linkEl).toHaveAttribute('target', '_blank');
  });
});
