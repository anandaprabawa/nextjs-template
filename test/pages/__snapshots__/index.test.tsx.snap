// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="flex flex-col items-center"
  >
    <h1
      class="text-4xl font-bold opacity-80"
    >
      NextJS Template
    </h1>
    <a
      class="uppercase text-sm py-2 px-4 border-blue-500 border rounded-3xl mt-8 text-blue-500 font-medium hover:bg-blue-50"
      href="https://github.com/anandaprabawa/nextjs-template"
      rel="noreferrer"
      target="_blank"
    >
      Go to repository
    </a>
  </main>
</DocumentFragment>
`;
